samples/c/text_generation/CMakeLists.txt

```diff
@@ -28,7 +27,7 @@ set (SAMPLE_LIST
     chat_sample_c
     benchmark_genai_c)
 
-foreach(sample ${SAMPLE_LIST})
+foreach(sample IN LISTS SAMPLE_LIST)
```

Comment:
It seems that using `foreach(sample IN LISTS SAMPLE_LIST)` to handle `SAMPLE_LIST` as `LISTS` is more robust: https://cmake.org/cmake/help/latest/command/foreach.html
Can we apply the same change cpp sample CMakeLists.txt too? 
https://github.com/openvinotoolkit/openvino.genai/blob/ab0bdcc03dc52da733267755d584cd48ab762cf1/samples/cpp/text_generation/CMakeLists.txt#L33

Reply:
yes, in a separate PR.